### PR TITLE
Isolate matter prerequisites 

### PR DESCRIPTION
**Pull request by hiltonlima**
## Merge only when the current release branch is integrated into main

Backend PR link: https://github.com/project-chip/certification-tool-backend/pull/78

- Remove branch requirement to use update.sh script
- Create the test_collections and matter program folder
- Fixed typo (renamed file 1-install-dependencies.sh)
- Create a temporary matter setup script (it need to be defined into matter_program_specific repo)

### CODE REVIEW

File: scripts/update.sh [hunk 1]
```diff
@@ -27,19 +27,6 @@ if [ $# -eq 1 ]; then
...
-if [ "$ROOT_BRANCH" = "HEAD" ]; then
-    echo "The HEAD is detached from a branch. Should it checkout to develop before proceeding?"
-    select yn in "Yes" "No"; do
-        case $yn in
-            Yes ) ROOT_BRANCH=develop; break;;
-            No ) echo "Aborting..."; exit;;
-        esac
-    done
-fi
-
```
Comment (hiltonlima) [inline scripts/update.sh:42]: Removed because we want to use git TAGs instead branches... Also we don't have developer branch on the repo.

**Review by ccruzagralopes — APPROVED** [review]

**Review by antonio-amjr — APPROVED** [review]

**Review by rquidute — APPROVED** [review]

**Review by ccruzagralopes — APPROVED** [review]

File: scripts/update.sh [hunk 2]
```diff
@@ -44,10 +44,18 @@ cd $ROOT_DIR
...
+set +e
 newgrp docker << END
-# You can do more lines than just this./
 docker compose pull
 END
+# Check if the docker images download failed
+# In case of failure, the images will be built locally
+if [ $? -ne 0 ]; then
+    newgrp docker << END
+    docker compose up --no-start
```
Comment (ccruzagralopes) [inline scripts/update.sh:55]: Is this enough? Or should the build script be called? I believe that script has some additional steps, but I'm not 100% sure

Comment (hiltonlima) [reply scripts/update.sh:55]: In the tests I did here, yes. The UP command is used to start the container and if the image does not exist, it builds one. The argument --no-start we give is that he didn't start the services at that time.

Comment (hiltonlima) [reply scripts/update.sh]: Now we are using build scripts. Thanks for it :)

**Pull request merged**